backend/open_webui/utils/utils.py

```diff
@@ -75,10 +73,15 @@ def get_http_authorization_cred(auth_header: str):
     except Exception:
         raise ValueError(ERROR_MESSAGES.INVALID_TOKEN)
 
+def get_api_key_auth_config():
+    from open_webui.config import ENABLE_API_KEY_AUTH
+    return ENABLE_API_KEY_AUTH
+
 
 def get_current_user(
     request: Request,
     auth_token: HTTPAuthorizationCredentials = Depends(bearer_security),
+    api_key_auth_enabled: bool = Depends(get_api_key_auth_config)
 ):
```

Comment:
I would like to comment a bit about the approach made in here:

The previous version of this functionality [used the `AppConfig` which is made available using the `Requests` instance.](https://github.com/antpyykk-kone/open-webui/blob/a9ccac4ffd049e06e575b50195b35c8acfafdf17/backend/open_webui/utils/utils.py#L91)

However, since the utilities for fetching the current user are used by all applications, we would have had to place the configuration option into all the applications. There could also be an easy chance for a bug if a new application is introduced and it forgets to set the configuration option (resulting in a KeyError).

---

We instead import the configuration option itself.
The reason why we use a lazy import, is because importing `open_webui.config` seems to normally result in cyclical import.

```sh
Error: cannot import name 'ENABLE_API_KEY_AUTH' from partially initialized module 'open_webui.config' (most likely due to a circular import) (/home/antpyykk/repos/open-webui/backend/open_webui/config.py
```

I was unfortunately not able to resolve this.